server/middlewares/roles.js

```diff
@@ -3,13 +3,13 @@ import httpStatus from 'http-status'
 import { APIError } from '../helpers/errors'
 
 function checkRole(checkVisitor, checkParticipant, checkOwner, req, res, next) {
-  if (!req.user) {
+  if (!req.user && req.freeFestival !== true) {
```

Comment:
I would not recommend putting this here, as this middleware is about permission roles and they should be defined but what role one has and not if a festival setting is set, this might lead to other parts of the application suddenly being open to the public ..